src/picongpu/include/particles/pusher/particlePusherFree.hpp

```diff
@@ -42,7 +43,7 @@ namespace picongpu
             {
 
                 Velocity velocity;
-                const PosType vel = velocity(mom, mass);
+                const float3_X vel = velocity(mom, mass);
```

Comment:
for momentum, `float3_X` is pretty common. but since the functor would in principle also support an other type, you would make it more general that way (else one would need to remove the template because it suggests it would support that interface and then explicitly casts).